hopp/simulation/technologies/dispatch/hybrid_dispatch.py

```diff
@@ -71,147 +63,51 @@ def _create_parameters(hybrid):
-        hybrid.trough_generation = pyomo.Var(
-            doc="Power generation of CSP trough [MW]",
-            domain=pyomo.NonNegativeReals,
-            units=u.MW,
-            initialize=0.0)
-        hybrid.trough_load = pyomo.Var(
-            doc="Load of CSP trough [MW]",
-            domain=pyomo.NonNegativeReals,
-            units=u.MW,
-            initialize=0.0)
-        self.power_source_gen_vars[t].append(hybrid.trough_generation)
-        self.load_vars[t].append(hybrid.trough_load)
-
-    def _create_trough_port(self, hybrid, t):
-        hybrid.trough_port = Port(initialize={'cycle_generation': hybrid.trough_generation,
-                                              'system_load': hybrid.trough_load})
-        self.ports[t].append(hybrid.trough_port)
-
-    def _create_battery_variables(self, hybrid, t):
-        hybrid.battery_charge = pyomo.Var(
-            doc="Power charging the electric battery [MW]",
-            domain=pyomo.NonNegativeReals,
-            units=u.MW,
-            initialize=0.0)
-        hybrid.battery_discharge = pyomo.Var(
-            doc="Power discharging the electric battery [MW]",
-            domain=pyomo.NonNegativeReals,
-            units=u.MW,
-            initialize=0.0)
-        self.power_source_gen_vars[t].append(hybrid.battery_discharge)
-        self.load_vars[t].append(hybrid.battery_charge)
-
-    def _create_battery_port(self, hybrid, t):
-        hybrid.battery_port = Port(initialize={'charge_power': hybrid.battery_charge,
-                                               'discharge_power': hybrid.battery_discharge})
-        self.ports[t].append(hybrid.battery_port)
+            units=u.dimensionless,
+        )
 
-    @staticmethod
-    def _create_grid_variables(hybrid, _):
-        hybrid.system_generation = pyomo.Var(
-            doc="System generation [MW]",
-            domain=pyomo.NonNegativeReals,
-            units=u.MW)
-        hybrid.system_load = pyomo.Var(
-            doc="System load [MW]",
-            domain=pyomo.NonNegativeReals,
-            units=u.MW)
-        hybrid.electricity_sold = pyomo.Var(
-            doc="Electricity sold [MW]",
-            domain=pyomo.NonNegativeReals,
-            units=u.MW)
-        hybrid.electricity_purchased = pyomo.Var(
-            doc="Electricity purchased [MW]",
-            domain=pyomo.NonNegativeReals,
-            units=u.MW)
-
-    def _create_grid_port(self, hybrid, t):
-        hybrid.grid_port = Port(initialize={'system_generation': hybrid.system_generation,
-                                            'system_load': hybrid.system_load,
-                                            'electricity_sold': hybrid.electricity_sold,
-                                            'electricity_purchased': hybrid.electricity_purchased})
-        self.ports[t].append(hybrid.grid_port)
-
-    def _create_grid_constraints(self, hybrid, t):
+    def _create_variables_and_ports(self, hybrid, t):
```

Comment:
I'd love to see a doc string for this to explain briefly how this goes about setting up the variables and ports for Pyomo